cmd/options/controller_options.go

```diff
@@ -24,15 +24,16 @@ import (
 
 // ControllerOptions contains options and configuration settings for the controller service.
 type ControllerOptions struct {
-	// ExtraVolumeTags is a map of tags that will be attached to each dynamically provisioned
-	// volume.
-	ExtraVolumeTags map[string]string
+	// ExtraTags is a map of tags that will be attached to each dynamically provisioned
+	// resource.
+	ExtraTags map[string]string
 	// ID of the kubernetes cluster. This is used only to create the same tags on volumes that
 	// in-tree volume volume plugin does.
 	KubernetesClusterID string
 }
 
 func (s *ControllerOptions) AddFlags(fs *flag.FlagSet) {
-	fs.Var(cliflag.NewMapStringString(&s.ExtraVolumeTags), "extra-volume-tags", "Extra volume tags to attach to each dynamically provisioned volume. It is a comma separated list of key value pairs like '<key1>=<value1>,<key2>=<value2>'")
+	fs.Var(cliflag.NewMapStringString(&s.ExtraTags), "extra-tags", "Extra tags to attach to each dynamically provisioned resource. It is a comma separated list of key value pairs like '<key1>=<value1>,<key2>=<value2>'")
+	fs.Var(cliflag.NewMapStringString(&s.ExtraTags), "extra-volume-tags", "DEPRECATED: Please use --extra-tags instead. Extra volume tags to attach to each dynamically provisioned volume. It is a comma separated list of key value pairs like '<key1>=<value1>,<key2>=<value2>'")
```

Comment:
If both flags are set, is it possible that the deprecated one will overwrite the other? Is there a way we can give `extra-tags` priority?

Perhaps we can:

1. Keep `ExtraVolumeTags` above
2. Keep the `driver.WithExtraVolumeTags` line in `cmd/main.go` 
3.  Keep the `WithExtraVolumeTags` in `driver.go` and change it to only set `o.extraTags` if it hasn't been set yet.

Does that makes sense?

Reply:
The other changes LGTM.

Reply:
Yeah that makes sense. Let me know if [this captures it](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/568/commits/9b993086ea47c565b6e945927879df3d809501a2).